http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/RunnableRequest.java

```diff
@@ -45,47 +55,58 @@ public RunnableRequest(RequestContext context) {
+                    makeRequest(context.nettyRequest());
+                } catch (Exception e) {
+                    handleFailure(() -> "Failed to make request to " + endpoint(), e);
+                }
             } else {
                 handleFailure(() -> "Failed to create connection to " + endpoint(), channelFuture.cause());
             }
         });
     }
 
-    @Override
-    public void abort() {
-        if (channel != null) {
-            channel.disconnect().addListener(ignored -> context.channelPool().release(channel));
-        }
+    /**
+     * Add any per-request handlers to the pipeline.
+     */
+    private void initializePerRequestHandlers() {
+        // Remove any existing handlers from the pipeline from the previous request.
+        removeIfExists(HttpStreamsClientHandler.class);
+        removeIfExists(ResponseHandler.class);
+
+        channel.pipeline().addLast(new HttpStreamsClientHandler());
+        channel.pipeline().addLast(new ResponseHandler());
     }
 
     /**
-     * Add handlers to enforce backpressure using the reactive pull model.
+     * Removes the handler from the pipeline if present.
      *
-     * @return Subscriber of outbound request data.
+     * @param handler Handler to remove, identified by class.
      */
-    private Subscriber<ByteBuffer> addBackpressureHandlers() {
-        final NettyHttpContentSubscriber httpContentSubscriber = new NettyHttpContentSubscriber(channel);
-        final Subscriber<ByteBuffer> byteBufferSubscriber = httpContentSubscriber.adapt();
-        HandlerPublisher<ByteBuffer> publisher = new CompletingHandlerPublisher(channel.eventLoop(), context.handler());
-        channel.pipeline().addLast(publisher, httpContentSubscriber);
-        channel.attr(PUBLISHER_KEY).set(publisher);
-        return byteBufferSubscriber;
+    private void removeIfExists(Class<? extends ChannelHandler> handler) {
+        if (channel.pipeline().get(handler) != null) {
+            channel.pipeline().remove(handler);
+        }
     }
 
-    private void makeRequest(HttpRequest request, Subscriber<ByteBuffer> subscriber) {
-        log.debug("Writing request: {}", request);
-        channel.write(request).addListener(wireCall -> {
-            if (!wireCall.isSuccess()) {
-                handleFailure(() -> "Failed to make request to " + endpoint(), wireCall.cause());
-            }
-        });
-        // Subscribe to the request Publisher to start requesting data to send
-        context.sdkRequestProvider().subscribe(subscriber);
+    @Override
+    public void abort() {
+        if (channel != null) {
+            channel.disconnect().addListener(ignored -> context.channelPool().release(channel));
+        }
+    }
+
+    private void makeRequest(HttpRequest request) {
+        log.debug("Writing request: " + request);
+        channel.writeAndFlush(new StreamedRequest(context.nettyRequest(), context.sdkRequestProvider(), channel))
```

Comment:
writeAndFlush, at low TPS this was buffering to a certain amount before sending the request resulting in higher latencies.